packages/medusa/src/repositories/image.ts

```diff
@@ -1,5 +1,35 @@
-import { EntityRepository, Repository } from "typeorm"
+import { EntityRepository, In, Repository } from "typeorm"
 import { Image } from "../models/image"
 
 @EntityRepository(Image)
-export class ImageRepository extends Repository<Image> {}
+export class ImageRepository extends Repository<Image> {
+  public async upsertImages(imageUrls: string[]) {
+    const existingImages = await this.find({
+      where: {
+        url: In(imageUrls),
+      },
+    })
+    const existingImagesMap = this.buildImagesMap_(existingImages)
+
+    const upsertedImgs: Image[] = []
+
+    for (const url of imageUrls) {
+      const aImg = existingImagesMap.get(url)
+      if (aImg) {
+        upsertedImgs.push(aImg)
+      } else {
+        const newImg = this.create({ url })
+        const savedImg = await this.save(newImg)
+        upsertedImgs.push(savedImg)
+      }
+    }
+
+    return upsertedImgs
+  }
+
+  private buildImagesMap_(images: Image[]): Map<string, Image> {
```

Comment:
comment: I really think the map instanciation should be done in the core of the method, having a separate method to instanciate an object would be like having a method to instanciate a boolean as well. wdyt?